Mod2/Lessons/Week4/CRUDwithORM.md

```diff
@@ -0,0 +1,182 @@
+};
+
+context.Plants.Add(pothos);
+context.SaveChanges();
+```
+
+  
+
+Adding a plant isn't much different than adding a room, except for two major differences.�
+1.  In the database, a Plant record has a room ID field, which serves as a foreign key to the Room table. On the other hand, we attach a plant to a Room object.    
+2.  Additionally, a Room contains a list of the Plants it houses. When we create a Plant, we also update the corresponding Room object.     
+
+In your breakout room, add a new plant to the bedroom in our application. It is a snake plant purchased on January 15, 2023. 
+
+## Read
+
+We have records in our database; specifically two rooms with a plant in each room. We can verify that by logging into pgAdmin and querying the PlantTracker database.
+  
+> What SQL statement would show all the rooms in your database? What about the plants?
+> `SELECT * FROM rooms`
+> `SELECT * FROM plants` 
+
+But what fun is that? We can read from the database using our application. Let's get information about the rooms using Entity Framework. Add the following code to the using block in the `Main` method in `Program.cs`:
+  
+```c#
+var rooms = from r in context.Rooms select r;
+foreach (var room in rooms)
+{
+	Console.WriteLine($"Room {room.Id}, Name: {room.Name},  Has Sunlight? {room.HasSunlight}");
+}
+```
+
+This code accomplishes the following:
+
+1.  Creates an array variable to hold all the rooms in the database.    
+2.  Iterates through each room using a foreach loop.    
+3.  For each room, display the ID, name, and if it has sunlight to the console.    
+
+In your breakout room, read and display all the information for each plant in the Plants table to the console. Use the previous Room example to give your output some "context".
+
+## Update
+
+We have tables and records in our database, and we can access them from our application. What if something changes? For example, we've installed blackout curtains in the bedroom and it no longer gets adequate sunlight. We need to update this information in our Rooms table.  
+
+> How would you update this room in your database using SQL statements?
+
+Let�s do the same using Entity Framework. Add the following code to the `Main` method to in `Program.cs`, :
+  
+```c#
+// UPDATE operation
+bedroom.HasSunlight = "false";
+context.SaveChanges();
+}
+```
+
+This code updates the value of HasSunlight to false in the bedroom object, then saves that change in the database. Add another foreach loop to see the change in the bedroom.
+
+```c#
+var rooms = from r in context.Rooms select r;
+foreach (var room in rooms)
+{
+	Console.WriteLine($"Room {room.Id}, Name: {room.Name},  Has Sunlight? {room.HasSunlight}");
+}
+```
+
+In your breakout room, update the type of your snake plant to a peace lily, then display all the information for each plant in the Plants table to the console. Don't be concerned if you named your variable SnakePlant or some other identifier that no longer makes sense.�
+
```

Comment:
This would be a great place to use LINQ methods to explore filtering/ordering data coming from the database!